crates/sui-snapshot/src/reader.rs

```diff
@@ -167,86 +168,79 @@ impl StateSnapshotReaderV1 {
                     .or_insert(BTreeMap::new())
                     .entry(current_part_num.try_into().unwrap())
                     .or_insert(hasher.finalize().digest);
+                all_files.insert((*bucket, current_part_num.try_into().unwrap()));
             }
         }
+
         let input_files: Vec<_> = self
             .object_files
             .iter()
             .flat_map(|(bucket, parts)| {
-                let vec: Vec<_> = parts.iter().map(|entry| (bucket, entry)).collect();
+                let vec: Vec<_> = parts
+                    .iter()
+                    .map(|entry| {
+                        all_files.remove(&(*bucket, *entry.0));
+                        (bucket, entry)
+                    })
+                    .collect();
                 vec
             })
             .collect();
+        all_files.is_empty().then_some(()).ok_or(anyhow!(
+            "Number of partitions in ref file does not match with number of object files"
+        ))?;
         let epoch_dir = self.epoch_dir();
         let remote_object_store = self.remote_object_store.clone();
         let indirect_objects_threshold = self.indirect_objects_threshold;
         let download_concurrency = self.concurrency;
-        let sha3_digests_cloned = sha3_digests.clone();
-        if let Err(e) = Abortable::new(
+        Abortable::new(
             async move {
-                let results: Vec<Result<(), anyhow::Error>> =
-                    futures::stream::iter(input_files.iter())
-                        .map(|(bucket, (part_num, file_metadata))| {
-                            let backoff = backoff::ExponentialBackoff::default();
-                            let epoch_dir = epoch_dir.clone();
-                            let file_path = file_metadata.file_path(&epoch_dir);
-                            let remote_object_store = remote_object_store.clone();
-                            let sha3_digests = sha3_digests_cloned.clone();
-                            async move {
-                                let bytes = retry(backoff, || async {
```

Comment:
Do we no longer need the retry logic?

Reply:
no because recently we baked retry logic in the `get` call, so it is redundant here